### PR TITLE
[mod_kazoo] Coverity fixes

### PR DESCRIPTION
**Pull request by jakubkarolczyk**

### CODE REVIEW

File: src/mod/event_handlers/mod_kazoo/kazoo_message.c [hunk 1]
```diff
@@ -251,13 +253,13 @@ cJSON * kazoo_event_add_field_to_json(cJSON *dst, switch_event_t *src, kazoo_fie
 			expanded = kz_event_expand_headers(src, field->value);
 			if(expanded != NULL && !zstr(expanded)) {
 				item = kazoo_event_add_json_value(dst, field, field->as ? field->as : field->name, expanded);
-				if(expanded != field->value) {
-					free(expanded);
-				}
 			}
+
+			switch_safe_free(expanded);
```
Comment (lazedo) [inline src/mod/event_handlers/mod_kazoo/kazoo_message.c:258]: this is wrong

**Review by lazedo — CHANGES_REQUESTED** [review]

File: src/mod/event_handlers/mod_kazoo/kazoo_message.c (hunk 1 above)
Comment (jakubkarolczyk) [reply src/mod/event_handlers/mod_kazoo/kazoo_message.c:258]: why you think this is wrong?

Comment (jakubkarolczyk) [reply src/mod/event_handlers/mod_kazoo/kazoo_message.c:258]: ok, I see

**Review by andywolk — APPROVED** [review]

**Comment by ivanuschak** [comment]
It looks like this fix causes crash on data kazoo exchange (on the service startup when FreeSwitch receives XML data from kazoo).
When the memory for the string `xml_str` is already freed [at this line](https://github.com/signalwire/freeswitch/blob/master/src/mod/event_handlers/mod_kazoo/kazoo_node.c#L1151), [another thread](https://github.com/signalwire/freeswitch/blob/master/src/mod/event_handlers/mod_kazoo/kazoo_fetch_agent.c#L315) is using this string to parse XML, and it causes crash when FreeSwitch is trying to parse XML from this released memory 

`2023-10-30 09:30:37.017722 98.37% [DEBUG] kazoo_fetch_agent.c:369 Sending configuration XML request (dec5f565-3108-48ae-9d38-1af699b210d0) to ecallmgr@ippbx.dev.esrp.or.nga911.com <1.3411.94>
2023-10-30 09:30:37.017722 98.37% [DEBUG] kazoo_node.c:1548 Sent erlang message to ecallmgr@ippbx.dev.esrp.or.nga911.com <1.3411.94>
2023-10-30 09:30:37.017722 98.37% [DEBUG] kazoo_fetch_agent.c:444 Received configuration XML (dec5f565-3108-48ae-9d38-1af699b210d0) after 0ms: ºú{𿊲2023-10-30 09:30:37.017722 98.37% [ERR] switch_xml.c:1800 Error[[error near line 1]: root tag missing]`

**Pull request merged**